docs/contributor_guide/release.rst

```diff
@@ -0,0 +1,39 @@
+.. release_guide
+
+Release Process
+---------------
+
+This page outlines the process for creating a new Fairlearn release.
+The following steps must be performed on a clone of Fairlearn, not
+a fork.
+
+#. Create an annotated tag using the command at the branch point:
+
+    :code:`git tag -a v<x.y.z> -m "Branch point for v<x.y.z> release"`
+
+#. Push the tag to GitHub:
+
+    :code:`git push origin v<x.y.z>`
+
+#. Create the branch for the release:
+
+    :code:`git checkout -b release/v<x.y.z>`
+
+#. On the release branch:
+
+    #. Update the version in `__init__.py` to `x.y.z`
+    #. Update the version in the ReadMe
```

Comment:
These should be PRs to that branch. Why do we have the version in the readme? :D

Reply:
Whether or not they're PRs depends on whether you make the changes before or after publishing the branch :-)

The ReadMe currently has a "The current stable release is ...." sentence, which is what needs updating.

Reply:
> The current stable release is ....

I mean I don't think we need such an information in the README, but not related to this PR I guess.

> Whether or not they're PRs depends on whether you make the changes before or after publishing the branch :-)

Ideally "creating a branch" would mean
- create a branch
- push

and nothing else. This way it's easier to keep track of the changes through PRs.

Reply:
Fair point on doing the PR on the release branch. I have tweaked the steps to reflect that.